src/transform/src/lib.rs

```diff
@@ -586,6 +586,8 @@ impl Optimizer {
             // identical. Check the `threshold_elision.slt` tests that fail if
             // you remove this transform for examples.
             Box::new(crate::threshold_elision::ThresholdElision),
+            // We need this to ensure that `CollectIndexRequests` gets a normalized plan.
```

Comment:
@ggevay  In the next PR (whataver that is), can you expand this comment to point out that in this specific case (#21175) we have an example where `FoldConstants` breaks the normalized form.

Reply:
Yes, will do!